scripts/migrate.js

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env node
+
+(async () => {
+  const {promisify} = require('util');
+  const {readdir} = require('fs');
+  const path = require('path');
+  const { createClient } = require('contentful-management');
+  const {default: runMigration} = require('contentful-migration/built/bin/cli');
+
+  // utility fns
+  const getVersionOfFile = (file) => file.replace('.js', '').replace(/_/g, '.');
+  const getFileOfVersion = (version) => version.replace(/\./g, '_') + '.js';
+
+  //
+  // Configuration variables
+  //
+  const [,, SPACE_ID, ENVIRONMENT_ID, CMA_ACCESS_TOKEN] = process.argv;
+  const LOCALE = 'en-US';
+  const MIGRATIONS_DIR = path.join('.', 'migrations');
+
+  const client = createClient({
+    accessToken: CMA_ACCESS_TOKEN
+  });
+  const space = await client.getSpace(SPACE_ID);
+  const environment = await space.getEnvironment(ENVIRONMENT_ID);
+  const readdirAsync = promisify(readdir);
+
+  console.log('Running with the following configuration');
+  console.log(`SPACE_ID: ${SPACE_ID}`);
+  console.log(`ENVIRONMENT_ID: ${ENVIRONMENT_ID}`);
+
+  // ---------------------------------------------------------------------------
+  // 1. Read all the available migrations from the file system
+  const availableMigrations = (await readdirAsync(MIGRATIONS_DIR))
+    .filter(file => /^\d+?_\d+?_\d+?\.js$/.test(file))
```

Comment:
The more I think about the worse of an idea I think it is to use semher here. /cc @Shy 

Since we're only storing the last ran version, not a list of all migrations we ever performed, we can't actually support any of the things why you'd want to to do with semver (e.g. release a patch 5.5.1 after releasing 5.6.0) nor do environments currently have any concept of a production environment except for the master one, which makes semver unnecessary.

I think this just raises expectations we can't meet.

Reply:
That's fair... simple numbers work as good. I'd like to post pone discussions around these though to get this first step in. :)

Reply:
Originally part of why I stored the version inside the migration files way back in the first draft of this repo was to be able to track the revision path. That way you'd be able to jump into the expected version migration file and backtrack through the previous files to have the whole upgrade tree. Similar to how Alembic (python DB migration tool) handles things: http://alembic.zzzcomputing.com/en/latest/cookbook.html#building-uptodate. 

It never got around to implementing it that way though and I now think it's not necessary. If you'd want to rebuild the contentful space from scratch for whatever reason I think you'd be better of using the import-export tooling rather than the migration CLI. 

I agree with you both right now that semver is overkill with how this example has evolved. We can cut it and replace it with simple numbers.